crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -155,17 +159,30 @@ impl CheckpointExecutor {
                     highest_executed = Some(checkpoint);
                 }
                 // Check for newly synced checkpoints from StateSync.
-                Ok(checkpoint) = self.mailbox.recv() => {
-                    debug!(
-                        sequence_number = ?checkpoint.summary.sequence_number,
-                        "received checkpoint summary from state sync"
-                    );
-                    SystemTime::now().duration_since(checkpoint.summary.timestamp())
-                        .map(|latency|
-                            self.metrics.checkpoint_contents_age_ms.report(latency.as_millis() as u64)
-                        )
-                        .tap_err(|err| warn!("unable to compute checkpoint age: {}", err))
-                        .ok();
+                received = self.mailbox.recv() => match received {
+                    Ok(checkpoint) => {
+                        debug!(
+                            sequence_number = ?checkpoint.summary.sequence_number,
+                            "received checkpoint summary from state sync"
+                        );
+                        SystemTime::now().duration_since(checkpoint.summary.timestamp())
+                            .map(|latency|
+                                self.metrics.checkpoint_contents_age_ms.report(latency.as_millis() as u64)
+                            )
+                            .tap_err(|err| warn!("unable to compute checkpoint age: {}", err))
+                            .ok();
+                    },
+                    // In this case, messages in the mailbox have been overwritten
+                    // as a result of lagging too far behind.
+                    Err(RecvError::Lagged(num_skipped)) => {
+                        debug!(
+                            "Checkpoint Execution Recv channel overflowed {:?} messages",
```

Comment:
Can we update the message here to indicate that this is benign? So we don't cause any red herring debugging down the line

Reply:
Given that it's already a debug log, I think it's sufficiently benign?

Reply:
Fair enough